### PR TITLE
[Updated] Trader prototype

### PR DESCRIPTION
**Pull request by sboigelot**
Add an XML definition for traders and read it in the World.cs

This is part on the ongoing addition I'm making to the trade system. 
To be integrated later with a gameevent and rigth click (PR #514) on the trader that will come to the landing pad to initiate a trade job.

sample
image
`

image

### CODE REVIEW

**Comment by ComputerOverlord** [comment]
I don't think an NPC is a good way to do it. I like my system, you use a furniture to buy stuff with a currency I called KR

**Comment by Mizipzor** [comment]
I added [Conflict], remove when youve fixed. :)

**Comment by sboigelot** [comment]
@ComputerOverlord I think both can exist at the same time as you suggested. I would love to see a furniture that allow you to order goods for KR :) Like a vending machine, or a space-ebay ^^

**Comment by sboigelot** [comment]
Fixes all conflict + added trader prototype for trader variety

image

**Comment by ComputerOverlord** [comment]
@sboigelot Good point. It'd be like a Government-based Space EBay... I guess :P

**Comment by sboigelot** [comment]
fixed new conflicts

**Comment by RodRitter** [comment]
Tested. Good job. Merged.

File: Assets/Scripts/Models/Trade/TraderPrototype.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        Trader t = new Trader
+        {
+            CurrencyBalance = Random.Range(MinCurrencyBalance, MaxCurrencyBalance),
+            Name = PotentialNames[Random.Range(PotentialNames.Count, PotentialNames.Count - 1)],
+            SaleMarginMultiplier = Random.Range(MinSaleMarginMultiplier, MaxSaleMarginMultiplier)
+        };
+
+        foreach (var potentialStock in PotentialStock)
+        {
+            var itemIsInStock = Random.Range(0f, 1f) > potentialStock.Rarity;
```
Comment (mmooxy) [inline Assets/Scripts/Models/Trade/TraderPrototype.cs]: Quote from [CONTRIBUTING.md](https://github.com/TeamPorcupine/ProjectPorcupine/blob/master/CONTRIBUTING.md#style-guidelines)
> Avoid using 'var', even when the type would be clear from context. Verbose typing is best typing.

**Pull request merged**